### ISSUE TITLE
Harp broken in Node 5/npm 3

### ISSUE THREAD

**Issue by omichelsen** [sintaxi/harp#510]
My harp installation stopped working after upgrade to Node 5 with the following message:

```
Error: The `libsass` binding was not found in /usr/local/lib/node_modules/harp/node_modules/terraform/node_modules/node-sass/vendor/darwin-x64-47/binding.node
This usually happens because your node version has changed.
Run `npm rebuild node-sass` to build the binding for your current node version.
```

The `rebuild` command did nothing, so I tried reinstalling Harp, which led to this epic fail:

```
gyp info spawn args [ 'V=1', 'BUILDTYPE=Release', '-C', 'build' ]
  c++ '-DNODE_GYP_MODULE_NAME=binding' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DBUILDING_NODE_EXTENSION' -I/Users/olem/.node-gyp/5.0.0/include/node -I/Users/olem/.node-gyp/5.0.0/src -I/Users/olem/.node-gyp/5.0.0/deps/uv/include -I/Users/olem/.node-gyp/5.0.0/deps/v8/include -I../../nan  -Os -gdwarf-2 -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++0x -fno-rtti -fno-exceptions -fno-threadsafe-statics -std=c++11 -MMD -MF ./Release/.deps/Release/obj.target/binding/src/binding.o.d.raw  -c -o Release/obj.target/binding/src/binding.o ../src/binding.cpp
In file included from ../src/binding.cpp:3:
../src/sass_context_wrapper.h:8:10: fatal error: 'sass_context.h' file not found
#include <sass_context.h>
         ^
1 error generated.
make: *** [Release/obj.target/binding/src/binding.o] Error 1
```

No variation of `sudo` or simliar will make it work. It obviously is caused by something in `node-sass` (again), but do you have any insights into an upcoming fix? I am not really able to run any of my projects without a larger effort to downgrade Node :-/

**Comment by isaacrg** [comment]
@omichelsen The problem appears to be from those logs specifically caused by the NPM package [`node-sass`](https://www.npmjs.com/package/node-sass). I have had similar issues, when trying to harp or harp-dependent applications, including on windows(alexgleason/harp-gui#22).

Some support may be available at [github.com/sass/node-sass/blob/master/TROUBLESHOOTING.md](https://github.com/sass/node-sass/blob/master/TROUBLESHOOTING.md), though if you continue to have these errors, try seeing if node-sass on its own will work.

If node-sass  wont work when you try installing and running it, then you may well wish to raise an issue [here](https://github.com/sass/node-sass/issues).

**Comment by omichelsen** [comment]
Thanks for the suggestions, but `npm install -g node-sass` works fine. It's only when installing Harp things go awry.

**Comment by isaacrg** [comment]
@sintaxi may well need a look at this, then

**Comment by james89** [comment]
@isaacrg Experiencing @omichelsen's error as well! Thanks

**Comment by james89** [comment]
@omichelsen If you're using homebrew, switching global node version back to v4.2.2 allowed me to install harp correctly

**Comment by silentrob** [comment]
Harp support is tied pretty closly with node-sass. I suspect when node-sass works on npm 3/node 5 harp will get bumped too. For the time being go back to node 4 and re-install. Hopefully you are using `nvm` and that is an easy task.

**Comment by robclancy** [comment]
@silentrob  Considering `node-sass` installs just fine I don't see how it can be the issue.

**Comment by isaacrg** [comment]
Harp might not be using the latest node-sass version, which could be the
difference.

**Comment by isaacrg** [comment]
I can confirm that heroku is running Node V5.0.0, and when I tried loading an app dependent on the latest version of harp on (0.19.0), node-sass wouldn't install.
Log at: http://pastebin.com/hDByf4yQ

**Comment by kennethormandy** [comment]
This will be fixed in the next release of Harp. If you need to get around it in the meantime, I’ve updated Terraform, Harp’s preprocessing engine, with the latest release of Node-sass. So, you can install that manually in the meantime if you’d like. In your project directory, you can run:

```sh
npm install --save-dev harp terraform@0.13.1
```

Or use [`npm shinkwrap`](https://docs.npmjs.com/cli/shrinkwrap) and set your own versions. We’ll have a new release out soon with this updated properly, though. Thanks for your patience! #502

**Comment by isaacrg** [comment]
Thanks!

**Comment by IngwiePhoenix** [comment]
Yup, broken here too. C++ errors arise during the build, reflecting the major native API changes, most of them being breaking ones.

Harp should be updated to reflect these changes too, imho. :)

**Comment by fustkilas** [comment]
@kennethormandy could you please expand on how to use shrinkwrap in this case?

Also, installing terraform and harp locally as you suggested worked, but I no longer have a global harp command available - how would I run harp?

**Comment by saper** [comment]
@fustkilas Try `./node_modules/.bin/harp`

**Comment by fustkilas** [comment]
@saper this returns me to the prompt, silently. nothing happens.

**Comment by fustkilas** [comment]
I ended up using https://github.com/tj/n and node 4.0.0, awaiting latest harp.

**Comment by aviris** [comment]
Deleted all node and npm traces from my system, reinstalled Node/npm 6.5.0/3.10.3 with Homebrew, then npm install -g harp@next

Works perfectly!

**Comment by adeonir** [comment]
Here I did not have to uninstall node/npm, only installed using harp@next
